contrib/validate-odm/validate-odm.sh

```diff
@@ -0,0 +1,481 @@
+
+  echo -n "    ▪ Check RuleSet test result in DSR:  "
+  diff=$(diff <(echo ${curl_result} | jq -S .) <(jq -S . $3))
+  [[ "${diff}" == "" ]] && echo_success "SUCCEEDED" || error "FAILED" "There are differencies with the expected response :\n${diff}" 1
+}
+
+#===========================
+# Function to delete a RuleApp deployment in RES
+# Globals:
+# - RES_URL
+# Arguments:
+# - $1 ruleApp name
+# - $2 ruleApp version
+# Outputs:
+# - writes deletion status
+
+function deleteRuleApp {
+  ruleapp_name=$1
+  ruleapp_version=$2
+  echo -n "    ▪ Delete RuleApp ${ruleapp_name}/${ruleapp_version} in RES:  "
+  ruleapps_result=$(curlRequest DELETE ${RES_URL}/res/api/v1/ruleapps/${ruleapp_name}/${ruleapp_version}) || error "ERROR $?" "${ruleapps_result}" $?
+
+  echo_success "DONE"
+}
+
+#===========================
+# Function to clean after test
+# Clean all ruleApps from a given deployment list
+# Arguments:
+# - $1 deployment information
+# Outputs:
+# - writes logs
+
+function clean {
+  if [[ ${CLEAN} ]]; then
+    echo "🗑️   Cleaning ..."
+    deploymentsList=$1
+    # Clean ruleApps in RES
+    echo "${deploymentsList}" | jq -r '.[] | .id + " " + .ruleAppName + " " + .ruleAppVersion' | while read deploymentId ruleAppName ruleAppVersion; do
+      deleteRuleApp ${ruleAppName} ${ruleAppVersion}
+    done
+  fi
+}
+
+function main {
+  # Scenario:
+  # ------------------------------
+  # 1. Import a decision service
+  # 2. Run a test suite
+  # 3. Deploy the RuleApps
+  # 4. Verify the deployment in RES
+  # 5. Execute a RuleSet in DSR
+  # 6. [Optional] Delete the RuleApps in RES
+
+  authArgs=()
+  parse_args "$@"
+  setAuthentication
+
+  filename="Loan_Validation_Service.zip"
+  # Download Loan_Validation_Service.zip if it does not exist or is not a valid zip file
+  unzip -q -t ${filename} &> /dev/null
+  if [[ $? != 0 ]]; then
+    echo -n "$(date) - ### Loan_Validation_Service.zip does not exist locally. Downloading...  "
+    url="https://github.com/DecisionsDev/odm-for-dev-getting-started/blob/master/Loan%20Validation%20Service.zip?raw=1"
+    wget -q ${url} -O ${filename} && echo_success "DONE" || error "ERROR" "Error downloading file from ${url}" $?
+  fi
+
+  importDecisionService ${filename}
+  decisionServiceId=$(getDecisionServiceId "Loan Validation Service") || error "ERROR" "${decisionServiceId}" $?
+
+  runTestSuite "${decisionServiceId}" "Main Scoring test suite"
+
+  deploymentsList=$(getDeploymentInfo ${decisionServiceId}) || error "ERROR" "${deploymentsList}" $?
+  echo "${deploymentsList}" | jq -r '.[] | .id + " " + .ruleAppName + " " + .ruleAppVersion' | while read deploymentId ruleAppName ruleAppVersion; do
+    deployRuleApp ${deploymentId} ${ruleAppName} ${ruleAppVersion}
+    verifyRuleApp ${ruleAppName} ${ruleAppVersion}
+  done
+
+  testRuleSet production_deployment/1.0/loan_validation_production/1.0 test-ruleset/loan_validation_test.json test-ruleset/loan_validation_test_response.json
```

Comment:
That's really weird to have a loop to deploy rule apps from DC deployments list and test a ruleset outside the loop with a hardcoded name... 
What's weird to me in that script is that it mixes global variables, functions arguments and hardcoded values making the entire logic not easy to understand. But maybe it would be better to discuss about this point rather than trying to explain in a review comment... :-/